lib/controller.js

```diff
@@ -50,37 +59,83 @@ function listTessels(options) {
-        tessel.logs.info('No Tessel\'s found');
+  return Promise.settle([usb.findConnections(),lan.findConnections()])
+    .reduce(function(connections, result){
+      // Check if the promises have been fulfilled
+      if(result.isFulfilled()){
+        // If so concat the results which should be an array
+        var connections = connections.concat(result.value());
+      }
+      return connections;
+    }, [])
+    .then(function(connections){
+      return new Promise(function(resolve, reject) {
+      if(!connections.length){
+        reject('No Tessels Found');
       }
-      return stuff;
+      resolve(connections);
+      });
+    })
+    // Sort by ip address
+    .then(function(connections){
+      return _.sortBy(connections, 'ip');
     })
-    // Iterate through each and look for a match on the provided serial
-    // number of the return the first Tessel if one wasn't provided
-    .reduce(function(memo, connection){
-      // if serial number matches connect
-      if (!opts.serial || opts.serial === connection.serialNumber) {
-        return connection;
+    // Filter by the options provided
+    .reduce(function(collector, connection){
+      // TODO: Refactor to use UUID once implemented
+
+      var match;
+      // Start with tessel selected by the command `tessel select`
+      if(process.env.SELECTED_TESSEL && process.env.SELECTED_TESSEL === connection.host ){
+        match = connection
+      }
+      // Over ride with command line options
+      // Match name
+      if(opts.name && opts[2] === connection.auth.host){
+        match = connection;
+      }
+
+      // Match ip
+      if(opts.ip && opts[2] === connection.ip){
+        match = connection;
       }
-      // if memo is not defined set memo to first tessel
-      if (!memo) return connection;
-      // If none of the above, pass memo along
-      return memo;
-    }, null)
-    .then(function(connection){
-      tessel.logs.info("Connected over", connection.connectionType + ".");
+      // If double match, warn user
+      // Return early with no change
+      if(match && collector.selected){
+        tessel.log.warn("Multiple matches found. Using tessel at IP address:" + connection.ip);
+        return collector;
+      }
+      // TODO: Put logic in for USB once implemented remove following case
+      if(connection.connectionType === 'USB'){
+        return collector;
+      }
+      // Assign selected to match
+      collector.selected = match;
+      // If the default hasn't been assigned yet, assign it
+      if(!collector.default){
+        collector.default = connection;
+      }
+      // Return with default and selected, if any
+      return collector;
+    }, {default: null, selected: null} )
+    .then(function(connections){
+      // Check if one was selected, if not, use default;
+      var connection = connections.selected ? connections.selected : connections.default;
+      var message = "Connected over" + connection.connectionType + "."
```

Comment:
You need to add a space between "Connected over" and `connection.connectionType`.